backend/de.metas.business/src/main/java/de/metas/order/compensationGroup/OrderGroupRepository.java

```diff
@@ -770,15 +766,19 @@ public GroupTemplateId getGroupTemplateId(@NonNull final GroupId groupId)
 	public I_C_OrderLine createRegularLineFromTemplate(
 			@NonNull final GroupTemplateRegularLine from,
 			@NonNull final I_C_Order targetOrder,
-			@Nullable final ConditionsId contractConditionsId)
+			final @NonNull RetrieveOrCreateGroupRequest request)
 	{
 		final I_C_OrderLine orderLine = orderLineBL.createOrderLine(targetOrder);
-		orderLine.setM_Product_ID(from.getProductId().getRepoId());
+		final ProductId productId = from.getProductId();
+		orderLine.setM_Product_ID(productId.getRepoId());
 		orderLine.setM_AttributeSetInstance_ID(AttributeSetInstanceId.NONE.getRepoId());
 		orderLine.setC_UOM_ID(from.getQty().getUomId().getRepoId());
-		orderLine.setQtyEntered(from.getQty().toBigDecimal());
+		orderLine.setQtyEntered(request.getQtyMultiplier().multiply(from.getQty().toBigDecimal()));
 		orderLine.setC_CompensationGroup_Schema_TemplateLine_ID(from.getId().getRepoId());
-		orderLine.setC_Flatrate_Conditions_ID(ConditionsId.toRepoId(contractConditionsId));
+		orderLine.setC_Flatrate_Conditions_ID(ConditionsId.toRepoId(request.getNewContractConditionsId()));
+		orderLine.setIsSkipInvoicing(from.isSkipInvoicing());
+
+		orderLine.setIsHideWhenPrinting(!productId.equals(request.getGroupingProductId()));
```

Comment:
hmm, that's not ok... repo shall be plain stupid and not carry on BL

Reply:
made as stupid as humanly possible.